llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -71,6 +71,11 @@ raw_ostream &llvm::operator<<(raw_ostream &OS, const VPValue &V) {
   V.print(OS, SlotTracker);
   return OS;
 }
+raw_ostream &llvm::operator<<(raw_ostream &OS, const VPRecipeBase &R) {
```

Comment:
Instead of adding this operator, you can instead replace the one above `operator<<(raw_ostem &OS, const VPValue &V)`. If this operator supports `VPRecipeBase` then that also removes the need for a cast in SLPVectorizer.cpp and the one you added in VPlanTest.cpp.

Reply:
Done.